### DISCUSSION TITLE
Difficulty sort with mods

### DISCUSSION THREAD

**Discussion by cloudiees** [ppy/osu#27483] [Ideas]
As someone who is trying to get into lazer the difficulty sort in song select is really frustrating while using mods like DT or HR as the maps sort by their unmodded star rating. Also the star rating slider has the same issue where it doesn't use the modded star rating, but the unmodded one. This would be a really nice feature to have. Or I'm just stupid and there is a setting I'm missing (if so pls tell me)

**Comment by uzervlad** [comment]
I'm not 100% sure, but I'll make an educated guess on why that might not be possible.

Stable [pre-calculates star ratings](https://github.com/ppy/osu/wiki/Legacy-database-file-structure#beatmap-information) for all mod combinations, which is pretty simple, considering the small amount of mods.

In lazer, on the other hand, DT alone has a hundred possible star rating values, nevermind something like DA, which is why lazer doesn't calculate star rating until it needs to be shown.
While storing all these values is *technically* possible, it would be extremely memory-inefficient and slow to calculate.
